File: RepositoryOld/HarbingerContext.fs
module Repository.HarbingerContext
open Microsoft.EntityFrameworkCore
open Repository.Message

type HarbingerContext(options: DbContextOptions<HarbingerContext>) =
    inherit DbContext(options)

    override __.OnModelCreating modelBuilder =
        modelBuilder.Entity<Message>()
            .Property(fun x -> x.Id)
            .HasDefaultValue("NEWID()")
        |> ignore
        
        
    override __.OnConfiguring optionsBuilder =
        optionsBuilder
            .UseNpgsql("Host=localhost;Database=harbinger;Username=postgres;Password=postgres")
        |> ignore

    [<DefaultValue>]
    val mutable message: DbSet<Message>
    member x.messages
        with get() = x.message
        and set v = x.message <- v
        
        
        
        

File: Repository/HarbingerContext.cs
using System;
using System.Linq;
using Microsoft.EntityFrameworkCore;

namespace Repository
{
    public class HarbingerContext : DbContext
    {
        public HarbingerContext()
        { }
        public HarbingerContext(DbContextOptions<HarbingerContext> options)
            : base(options)
        {
        }


        public DbSet<MessageDto> Messages { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseNpgsql("Host=localhost;Database=harbinger;Username=postgres;Password=postgres");
        }

        public override int SaveChanges()
        {
            var entities = ChangeTracker.Entries().Where(x => x.Entity is EntityBase && (x.State == EntityState.Added || x.State == EntityState.Modified));

            foreach (var entity in entities)
            {
                if (entity.State == EntityState.Added)
                {
                    ((EntityBase) entity.Entity).CreatedAt = DateTime.UtcNow;
                }

                ((EntityBase)entity.Entity).UpdatedAt = DateTime.UtcNow;
            }
            
            return base.SaveChanges();
        }
    }
}